### PR TITLE
user info is rendered by user id

### PR DESCRIPTION
**Pull request by Malik-Musleh**
Start with user profile the user info get rendered the back end edited to met the front end requirement but still under progress.

### CODE REVIEW

File: back-end/controllers/main-controller.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+const addProduct = (req, res) => {
+    const { store_id, product_name, product_descripition, quantity_per_unit, unit_price, available_product, picture } = req.body
+    const data = [store_id, product_name, product_descripition, quantity_per_unit, unit_price, available_product, picture]
+    const query = `INSERT INTO products (store_id,product_name,product_descripition,quantity_per_unit,unit_price,available_product,picture)
+VALUES (?,?,?,?,?,?,?) `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+const updateProduct = (req, res) => {
+    const { store_id, product_name, product_descripition, quantity_per_unit, unit_price, available_product, picture, product_id } = req.body
+    const data = [product_name, product_descripition, quantity_per_unit, unit_price, available_product, picture, product_id]
+    const query = `UPDATE products SET product_name=?,product_descripition=?,quantity_per_unit=?,
+        unit_price=?,available_product=?,picture=? WHERE product_id=${product_id} `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const getproducts = (req, res) => {
+    const query = `SELECT * from products WHERE store_id=?`
+    const data = [req.body.store_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const deleteProduct = (req, res) => {
+    const query = `DELETE FROM products WHERE product_id=?`
+    const data = [req.body.product_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+const addStore = (req, res) => {
+    const { store_name, store_category, store_pic, user_id } = req.body
+    const data = [store_name, store_category, store_pic, user_id]
+    const query = `INSERT INTO store (store_name,store_category,store_pic,user_id)
+    VALUES (?,?,?,?) `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+const updateStore = (req, res) => {
+    const { store_name, store_category, store_pic, store_id } = req.body
+    const data = [store_name, store_category, store_pic]
+    const query = `UPDATE store SET store_name=?,store_category=?,store_pic=? WHERE store_id=${store_id} `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const getStores = (req, res) => {
+    const query = `SELECT * from store WHERE user_id=?`
+    const data = [req.body.user_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    })
+}
+const deleteStore = (req, res) => {
+    const query = `DELETE FROM store WHERE store_id=?`
+    const data = [req.body.store_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const deleteItem = (req, res) => {
+    const query = `DELETE FROM items WHERE items_id=?`
+    const data = [req.body.item_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const getItems = (req, res) => {
+    const query = `SELECT * from items WHERE orders_id=?`
+    const data = [req.body.orders_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+const createOrder = (req, res) => {
+    const { user_id, delivary_user_id, store_id, item_id } = req.body
+    const data = [user_id, delivary_user_id, store_id, item_id]
+    const query = `INSERT INTO orders (user_id,delivary_user_id,store_id,item_id)
+    VALUES (?,?,?,?) `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const getOrders = (req, res) => {
+    const query = `SELECT * from orders WHERE user_id=?`
+    const data = [req.body.user_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/main-controller.js [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const deleteOrder = (req, res) => {
+    const query = `DELETE FROM orders WHERE order_id=?`
+    const data = [req.body.order_id]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
+const connection = require("../db")
+const bcrypt = require("bcrypt")
+const jwt = require("jsonwebtoken")
+const express = require('express');
+const app = express();
+require("dotenv").config()
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove these since they are not needed

File: back-end/controllers/users_controller.js [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+
+
+const register = async (req, res) => {
+    const { first_name, last_name, address, city, region, phone_number, email, password,
+        image_profile, payment_id, role_id, store_id } = req.body
+    console.log("SALT", process.env.SALT);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+    const hashedPassword = await bcrypt.hash(password, Number(process.env.SALT));
+    console.log(hashedPassword);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+    const data = [first_name, last_name, role_id, address, city, region, phone_number, email, hashedPassword,
+        image_profile, payment_id, store_id]
+    const query = `INSERT INTO users (first_name,last_name,role_id,address,city,region,phone_number,email,password,
+        image_profile,payment_id,store_id)
+    VALUES (?,?,?,?,?,?,?,?,?,?,?,?) `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+                };
+                const token = jwt.sign(payload, process.env.SECRET, options);
+                console.log(token);
+                res.json(token);
+            } else {
+                // res.status(422);
+                res.json({ error: "Invalid login check your password" });
+            }
+        } else {
+            // res.status(404);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: Add the correct status code or remove the commented out code

File: back-end/controllers/users_controller.js [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+};
+
+const getAllUsers = (req, res) => {
+    const query = `SELECT * FROM users WHERE is_deleted=?`
+    const data = [0]
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/controllers/users_controller.js [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+        res.json(results)
+    })
+}
+const getUserById = (req, res) => {
+    const query = `SELECT * FROM users WHERE user_id=${req.params.user_id}`
+    connection.query(query,  (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/users_controller.js]: remove the console log

File: back-end/db.js [hunk 21]
```diff
@@ -1,4 +1,5 @@
 const mysql = require('mysql2');
+require('dotenv').config();
```
Comment (Besslan) [inline back-end/db.js]: You do not need this over here, in the server it must be before the DB file require

File: back-end/schemas/schemas.sql [hunk 22]
```diff
@@ -1,21 +1,106 @@
...
-USE db_name;
+create database project_5;
+use project_5 
 
-CREATE TABLE users (
-    id INT AUTO_INCREMENT NOT NULL,
-   
-    PRIMARY KEY (id)
+--store
+
```
Comment (Besslan) [inline back-end/schemas/schemas.sql]: remove empty line

File: front-end/src/App.js [hunk 23]
```diff
@@ -1,29 +1,64 @@
...
+  const [email, setEmail] = useState("")
+  const [userPic, setUserPic] = useState("")
+  const [PhoneNumber, setPhoneNumber] = useState("")
+  const [Info, setInfo] = useState()
+  const getUser = async (infoArgumnt) => {
+    console.log(infoArgumnt)
+    axios
+      .get(`http://localhost:5000/users/${infoArgumnt}`)
+      .then(async (response) => {
+        console.log("response", response)
```
Comment (Besslan) [inline front-end/src/App.js]: remove the console log

File: p5.code-workspace [hunk 24]
```diff
@@ -0,0 +1,7 @@
+{
+    "folders": [
+        {
+            "path": "."
+        }
+    ]
+}
```
Comment (Besslan) [inline p5.code-workspace]: Add the `.cod-workspace` to`. gitignore `

File: back-end/controllers/main-controller.js [hunk 25]
```diff
@@ -0,0 +1,186 @@
...
+        res.json(results)
+    })
+}
+//********************orders */
+
```
Comment (Besslan) [inline back-end/controllers/main-controller.js:120]: remove the empty line

File: back-end/controllers/main-controller.js [hunk 26]
```diff
@@ -0,0 +1,186 @@
...
+const createItem = (req, res) => {
+    const { orders_id, product_id, total_price } = req.body
+    const data = [orders_id, product_id, total_price]
+    const query = `INSERT INTO items (orders_id,product_id,total_price)
+    VALUES (?,?,?) `
+    connection.query(query, data, (err, results) => {
+        if (err) {
+            console.log(err);
+        }
+        console.log(results);
```
Comment (Besslan) [inline back-end/controllers/main-controller.js]: remove the console log

**Comment by Malik-Musleh** [comment]
Back end updated to met the front end requirement but it still under editing

File: back-end/.gitignore [hunk 27]
```diff
@@ -1,2 +1,3 @@
 *.env
-node_modules
\ No newline at end of file
+node_modules
+p5
```
Comment (Besslan) [inline back-end/.gitignore]: First, add an empty line
second, add `.cod-workspace` instead as I mentioned earlier

File: back-end/controllers/main-controller.js [hunk 28]
```diff
@@ -75,35 +83,33 @@ const updateStore = (req, res) => {
         if (err) {
             console.log(err);
         }
-        console.log(results);
         res.json(results)
     })
 }
 
-const getStores = (req, res) => {
+const getStores = (req, res) => {
```
Comment (Besslan) [inline back-end/controllers/main-controller.js:89]: Remove the extra white space at the end of line 90

File: front-end/.gitignore [hunk 29]
```diff
@@ -1 +1,2 @@
-node_modules
\ No newline at end of file
+node_modules
+p5
```
Comment (Besslan) [inline front-end/.gitignore]: same as the other gitignore

File: back-end/schemas/schemas.sql [hunk 30]
```diff
@@ -32,6 +32,8 @@ CREATE table products(
 product_id int AUTO_INCREMENT NOT NULL ,
 category_id int ,
 store_id int ,
+item_id int ,
+
```
Comment (Besslan) [inline back-end/schemas/schemas.sql:36]: Remove this empty line

**Pull request merged**